externs/shaka/player.js

```diff
@@ -1011,3 +1025,28 @@ shaka.extern.PlayerConfiguration;
  * @exportDoc
  */
 shaka.extern.LanguageRole;
+
+
+/**
+ * @typedef {{
+ *   height: number,
+ *   left: number,
+ *   top: number,
+ *   uris: !Array.<string>,
+ *   width: number
+ * }}
+ *
+ * @property {number} height
+ *    The thumnnail height in percentage.
+ * @property {number} left
+ *    The thumnnail left position in percentage.
+ * @property {number} top
+ *    The thumnnail top position in percentage.
+ * @property {!Array.<string>} uris
+ *   An array of URIs to attempt.  They will be tried in the order they are
+ *   given.
+ * @property {number} width
+ *    The thumnnail width in percentage.
+ * @exportDoc
+ */
+shaka.extern.Thumbnail;
```

Comment:
The only thing I think might be missing here is the width/height of the thumbnail in pixels.  It's fine to take a percentage of the full sheet to get one tile, but if I don't know what the aspect ratio of the resulting thumbnail should be, do I have enough information to show this tile in my UI?  I think I would need either the tile size or the original sheet's full size to convert the percentage into a usable offset.

How about this as an alternative:

To render an image tile from a sprite sheet in CSS, I only need the width/height of the tile, plus the offset to the tile, all in pixels.  For example, with size and offset of the tile in pixels, this would work:

```js
div.style.backgroundImage = thumb.uris[0];
div.style.width = `${thumb.width}px`;
div.style.height = `${thumb.height}px`;
div.style.backgroundPositionX = `-${thumb.x}px`;
div.style.backgroundPositionY = `-${thumb.y}px`;
```

Reply:
I have changed the logic to use pixels, and the app is responsible for painting it as needed (either in pixel or percentage)